codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -69,61 +69,110 @@
     private final Http2Settings initialSettings;
     private ChannelFutureListener closeListener;
     private BaseDecoder byteDecoder;
-    private long gracefulShutdownTimeoutMillis = DEFAULT_GRACEFUL_SHUTDOWN_TIMEOUT_MILLIS;
+    private long gracefulShutdownTimeoutMillis;
 
-    public Http2ConnectionHandler(boolean server) {
-        this(server, true);
+    public static final class Builder extends BuilderBase<Http2ConnectionHandler, Builder> {
+        @Override
+        public Http2ConnectionHandler build0(Http2ConnectionDecoder decoder,
+                                             Http2ConnectionEncoder encoder) {
+            return new Http2ConnectionHandler(decoder, encoder, initialSettings());
+        }
     }
 
-    public Http2ConnectionHandler(boolean server, boolean validateHeaders) {
-        this(new DefaultHttp2Connection(server), validateHeaders);
-    }
+    public abstract static class BuilderBase<T extends Http2ConnectionHandler, B extends BuilderBase<T, B>> {
+        private Http2Settings initialSettings = new Http2Settings();
```

Comment:
Is this what you had in mind for https://github.com/netty/netty/pull/4299#discussion-diff-40876131?